src/sgame/sg_votes.cpp

```diff
@@ -0,0 +1,1198 @@
+								  : QQ( N_( "Team vote failed ($1$ of $2$; quorum not reached)" ) ),
+			level.team[ team ].voteYes + level.team[ team ].voteNo, level.team[ team ].numPlayers );
+	}
+	else if ( pass )
+	{
+		cmd = va( "print_tr %s %d %d",
+		          ( team == TEAM_NONE ) ? QQ( N_( "Vote passed ($1$ — $2$)" ) )
+		                                : QQ( N_( "Team vote passed ($1$ — $2$)" ) ),
+		          level.team[ team ].voteYes, level.team[ team ].voteNo );
+	}
+	else
+	{
+		cmd = va( "print_tr %s %d %d %.0f",
+		          ( team == TEAM_NONE ) ? QQ( N_( "Vote failed ($1$ — $2$; $3$% needed)" ) )
+		                                : QQ( N_( "Team vote failed ($1$ — $2$; $3$% needed)" ) ),
+		          level.team[ team ].voteYes, level.team[ team ].voteNo, votePassThreshold * 100 );
+	}
+
+	if ( team == TEAM_NONE )
+	{
+		trap_SendServerCommand( -1, cmd );
+	}
+	else
+	{
+		G_TeamCommand( team, cmd );
+	}
+
+	G_ResetVote( team );
+}
+
+void G_ResetVote( team_t team )
+{
+	int i;
+
+	level.team[ team ].voteTime = 0;
+	level.team[ team ].voteYes = 0;
+	level.team[ team ].voteNo = 0;
+	level.team[ team ].voted = 0;
+
+	for ( i = 0; i < level.maxclients; i++ )
+	{
+		level.clients[ i ].pers.voted &= ~( 1 << team );
+		level.clients[ i ].pers.voteYes &= ~( 1 << team );
+		level.clients[ i ].pers.voteNo &= ~( 1 << team );
+	}
+
+	trap_SetConfigstring( CS_VOTE_TIME + team, "" );
+	trap_SetConfigstring( CS_VOTE_STRING + team, "" );
+	trap_SetConfigstring( CS_VOTE_YES + team, "0" );
+	trap_SetConfigstring( CS_VOTE_NO + team, "0" );
+}
+
+/*
+==================
+G_Vote
+==================
+*/
+void G_Vote( gentity_t* ent, team_t team, bool voting )
+{
+	if ( !level.team[ team ].voteTime )
+	{
+		return;
+	}
+
+	if ( voting && ( ent->client->pers.voted & ( 1 << team ) ) )
+	{
+		return;
+	}
+
+	if ( !voting && !( ent->client->pers.voted & ( 1 << team ) ) )
+	{
+		return;
+	}
+
+	ent->client->pers.voted |= 1 << team;
+
+	// TODO maybe refactor vote no/yes in one and only one variable and divide the NLOC by 2 ?
+
+	if ( voting )
+	{
+		level.team[ team ].voted++;
+	}
+	else
+	{
+		level.team[ team ].voted--;
+	}
+
+	if ( ent->client->pers.voteYes & ( 1 << team ) )
+	{
+		if ( voting )
+		{
+			level.team[ team ].voteYes++;
+		}
+		else
+		{
+			level.team[ team ].voteYes--;
+		}
+
+		trap_SetConfigstring( CS_VOTE_YES + team, va( "%d", level.team[ team ].voteYes ) );
+	}
+
+	if ( ent->client->pers.voteNo & ( 1 << team ) )
+	{
+		if ( voting )
+		{
+			level.team[ team ].voteNo++;
+		}
+		else
+		{
+			level.team[ team ].voteNo--;
+		}
+
+		trap_SetConfigstring( CS_VOTE_NO + team, va( "%d", level.team[ team ].voteNo ) );
+	}
+}
+
+static constexpr char MARKER = '$';
+
+static bool IsEscapedMarker( Str::StringRef s, size_t pos )
+{
+	return pos + 1 < s.size() && s[ pos ] == MARKER && s[ pos + 1 ] == MARKER;
+}
+
+// Use a custom escape to avoid issues with Cmd::Escape unintentionally terminating quotes
```

Comment:
To make this clearer please mention that it needs to work in both quoted and unquoted contexts.

Reply:
noted